flyctl/app_config.go

```diff
@@ -54,9 +56,7 @@ func LoadAppConfig(configFile string) (*AppConfig, error) {
 		return nil, err
 	}
 
-	appConfig := AppConfig{
-		Definition: map[string]interface{}{},
-	}
+	appConfig := *NewAppConfig()
```

Comment:
Why are we dereferencing here?

Reply:
The method returns a pointer by default, which we actually don't want in this context. We should probably get the method to return a struct by default and let the caller take the address if they want.

Reply:
I think this is part of [`func LoadAppConfig(configFile string) (*AppConfig, error) {`](https://github.com/superfly/flyctl/blob/89efffb440a3512f80944c30cb1832ff82f0acb7/flyctl/app_config.go#L53) which returns a pointer, no? Sorry if I'm missing something obvious!

Reply:
Hi there again - let me clarify: I understand that in this particular context we used to do the following:

```
appConfig := AppConfig{
	Definition: map[string]interface{}{},
}
```

Changing from this to `appConfig := NewAppConfig()` (with `appConfig` ending up being a pointer) would not affect the rest of the code.

Sorry for the confusion!

Reply:
The last line (the return) was taking the address again – I've removed that now and made this a simple call to `NewAppConfig()`.